Check if running in NVIDIA graphics mode

crawfxrd opened the pull request:
Just checking for the proprietary drivers is no longer sufficient, as they will be loaded in hybrid graphics mode but not be used for rendering the desktop.

Rely on the `/etc/prime-discrete` file written by system76-power and gpu-manager to check which rendering mode is being used, falling back to checking the modules if it doesn't exist.

jacobgkau commented:
This is meant to fix https://github.com/pop-os/hidpi-daemon/issues/34, right?

crawfxrd commented:
Yeah, I guess. I did it to stop the spam on oryp10.

crawfxrd commented:
I can't dismiss the HiDPI notification, which is the same behavior I see while in NVIDIA graphics mode.

jacobgkau commented:
Confirmed this gets the HiDPI Daemon running again on addw2 in Hybrid mode. Still runs in NVIDIA and Integrated modes too.

I am seeing an issue in Hybrid mode that is not appearing in NVIDIA mode. An external LoDPI display (27" 1440p ViewSonic VP2771LED) always appears with 200% scaling, regardless of the mode that the daemon is in. In NVIDIA mode, the external display always shows at 100% scale.

Correct behavior in NVIDIA mode (external display on the right):

![Screenshot from 2022-08-30 16-15-59](https://user-images.githubusercontent.com/7199422/187553771-fd17c3ba-77ce-4c3e-9986-57b1eb0d65e5.jpg)

Same config in Hybrid mode (external display still on the right, although it defaulted to above multiple times):

![Screenshot from 2022-08-30 16-19-27](https://user-images.githubusercontent.com/7199422/187553779-0abb6e76-68fd-420a-823e-1d0dd08882aa.jpg)

I can't compare the behavior to Integrated mode since addw2 requires NVIDIA for any of the external display ports.

isantop reviewed (dismissed):
Code looks good on my end here, as long as QA passes :+1:

crawfxrd commented:
I have a Dell U2515H plugged in via HDMI. I see HiDPI overriding scaling to 200% on the eDP. And if I reboot with it plugged it, it thinks the external display is the default device.

crawfxrd committed:
Check if running in NVIDIA graphics mode

Just checking for the proprietary drivers is no longer sufficient, as
they will be loaded in hybrid graphics mode but not be used for
rendering the desktop.

Rely on the `/etc/prime-discrete` file written by system76-power and
gpu-manager to check which rendering mode is being used, falling back to
checking the modules if it doesn't exist.

Signed-off-by: Tim Crawford <tcrawford@system76.com>

crawfxrd committed:
Use `with` blocks for file access

Signed-off-by: Tim Crawford <tcrawford@system76.com>

crawfxrd committed:
Bump version to 18.04.7~~alpha

crawfxrd commented on debian/changelog:
```diff
@@ -1,3 +1,10 @@
+hidpi-daemon (18.04.7~~alpha) bionic; urgency=low
```
Should I change this to `focal`?

ahoneybun replied:
Updating the version and codename might be a good idea depending on how much longer we believe we need to include this software.
